py_ecc/fields/__init__.py

```diff
@@ -0,0 +1,160 @@
+from .field_elements import (
+    FQ,
+    FQP,
+    FQ2,
+    FQ12,
+)
+
+from .field_properties import (
+    field_properties,
+)
+
+from .optimized_field_elements import (
+    FQ as optimized_FQ,
+    FQP as optimized_FQP,
+    FQ2 as optimized_FQ2,
+    FQ12 as optimized_FQ12,
+)
+
+
+# Create seperate classes for all Fields for each curve
+
+bn128_FQ = type(
```

Comment:
Is there a reason we're using `type` here instead of normal class declaration?

```python
class bn128_FQ(FQ):
    curve_name = "bn128"
    field_modulus: field_properties["bn128"]["field_modulus"]
```

If there isn't, these should be converted to normal classes.